platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/ValueSanitizers.java

```diff
@@ -34,13 +34,27 @@ public Object sanitize(@Nullable Object value) {
             }
         }
     };
+
+    private static final ValueSanitizer<Object> IMPLICIT_CAST_TO_LONG_SANITIZER = new ValueSanitizer<Object>() {
+        @Override
+        @Nullable
+        public Object sanitize(@Nullable Object value) {
+            if (value instanceof Integer) {
```

Comment:
💭 We currently do only int -> long. I think others types are not often used with Gradle so we don't need to implicitly cast them